core/src/main/java/io/questdb/cutlass/pgwire/modern/PGConnectionContextModern.java

```diff
@@ -1483,6 +1457,31 @@ static short getShortUnsafe(long address) {
         return Numbers.bswap(Unsafe.getUnsafe().getShort(address));
     }
 
+    @Override
+    protected void doInit() {
+        sqlExecutionContext.with(getFd());
+
+        if (recvBuffer == 0) {
```

Comment:
send and recv buffers are freed in clear() call - the checks are redundant

Reply:
agree, it is likely that this could be removed.
there is no prealloc in PG context.

Reply:
tried to remove it, it breaks tests.
might be ok to remove it for prod code, it could be just tests not calling clear().

Reply:
replaced `if` statement with `assert` in https://github.com/questdb/questdb/pull/5934/commits/9a029ad900d5d7f7d88d96d341ee7936aac9d3b9